packages/angular_devkit/build_angular/src/builders/jest/index.ts

```diff
@@ -0,0 +1,147 @@
+      };
+    }
+
+    // Build all the test files.
+    const testFiles = await findTestFiles(options, context.workspaceRoot);
+    const jestGlobal = path.join(__dirname, 'jest-global.mjs');
+    const initTestBed = path.join(__dirname, 'init-test-bed.mjs');
+    const buildResult = await build(context, {
+      // Build all the test files and also the `jest-global` and `init-test-bed` scripts.
+      entryPoints: new Set(Array.from(testFiles).concat([jestGlobal, initTestBed])),
+      tsConfig: options.tsConfig,
+      polyfills: options.polyfills ?? ['zone.js', 'zone.js/testing'],
+      outputPath: testOut,
+      aot: false,
+      index: null,
+      outputHashing: OutputHashing.None,
+      outExtension: 'mjs', // Force native ESM.
+      commonChunk: false,
+      optimization: false,
+      buildOptimizer: false,
+      sourceMap: {
+        scripts: true,
+        styles: false,
+        vendor: true,
+      },
+    });
+    if (!buildResult.success) {
+      return buildResult;
+    }
+
+    // Execute Jest on the built output directory.
+    const jestProc = execFile('node', [
+      '--experimental-vm-modules',
+      jest,
+
+      `--rootDir=${testOut}`,
+      '--testEnvironment=jsdom',
+      '--color',
+
+      // Run basically all files in the output directory, any excluded files were already dropped by the build.
+      `--testMatch=${path.join('<rootDir>', '**', '*.mjs')}`,
+
+      // Load polyfills and initialize the environment before executing each test file.
+      // IMPORTANT: Order matters here.
+      // First, we execute `jest-global.mjs` to initialize the `jest` global variable.
+      // Second, we execute user polyfills, including `zone.js` and `zone.js/testing`. This is dependent on the Jest global so it can patch
+      // the environment for fake async to work correctly.
+      // Third, we initialize `TestBed`. This is dependent on fake async being set up correctly beforehand.
+      `--setupFilesAfterEnv=${path.join('<rootDir>', 'jest-global.mjs')}`,
+      `--setupFilesAfterEnv=${path.join('<rootDir>', 'polyfills.mjs')}`,
+      `--setupFilesAfterEnv=${path.join('<rootDir>', 'init-test-bed.mjs')}`,
+
+      // Don't run any infrastructure files as tests, they are manually loaded where needed.
+      `--testPathIgnorePatterns=${path.join('<rootDir>', 'jest-global\\.mjs')}`,
+      `--testPathIgnorePatterns=${path.join('<rootDir>', 'polyfills\\.mjs')}`,
+      `--testPathIgnorePatterns=${path.join('<rootDir>', 'init-test-bed\\.mjs')}`,
+
+      // Skip shared chunks, as they are not entry points to tests.
+      `--testPathIgnorePatterns=${path.join('<rootDir>', 'chunk-.*\\.mjs')}`,
+    ]);
+
+    // Stream test output to the terminal.
+    jestProc.child.stdout?.on('data', (chunk) => {
+      context.logger.info(chunk);
+    });
+    jestProc.child.stderr?.on('data', (chunk) => {
+      context.logger.info(chunk);
```

Comment:
Nit: This will cause stderr to be redirected to stdout
```suggestion
      context.logger.error(chunk);
```

Reply:
Yeah, I didn't like that either, but printing to stderr colors all the inputs red and clobbers Jest own coloring. Not sure if there's a better way to print to stderr while retaining the output style?

Reply:
There is no good way right now when using the logger. You'd need to use `process.stderr` directly.

Reply:
Updated to write to stderr with a comment explaining why `context.logger.error` is undesirable.